tools/python/mbed_tools/cli/cmsis_mcu_descr.py

```diff
@@ -0,0 +1,163 @@
+    name="cmsis-mcu-descr",
+    help="Manage CMSIS MCU description JSON file"
+)
+def cmsis_mcu_descr():
+
+    # Set up logger defaults
+    LOGGER.setLevel(logging.INFO)
+
+
+def open_cmsis_cache(must_exist=True) -> cmsis_pack_manager.Cache:
+    """
+    Open an accessor to the CMSIS cache.  Also prints how old the cache is.
+    """
+
+    cmsis_cache = cmsis_pack_manager.Cache(False, False)
+
+    index_file_path = pathlib.Path(cmsis_cache.index_path)
+    if not index_file_path.exists() and must_exist:
+        raise RuntimeError("CMSIS device descriptor cache does not exist!  Run 'python -m mbed_tools.cli.main cmsis-mcu-descr reload-cache' to populate it!")
+
+    # Check how old the index file is
+    index_age = humanize.naturaltime(datetime.datetime.fromtimestamp(index_file_path.stat().st_mtime))
+    LOGGER.info("CMSIS MCU description cache was last updated " + index_age)
+
+    return cmsis_cache
+
+
+def get_mcu_names_used_by_targets_json5() -> Set[str]:
+    """
+    Accumulate set of all `device_name` properties used by all targets defined in targets.json5
+    """
+    LOGGER.info("Scanning targets.json5 for used MCU names...")
+    used_mcu_names = set()
+    targets_json5_contents = decode_json_file(TARGETS_JSON5_PATH)
+    for mbed_target, target_details in targets_json5_contents.items():
+        if "device_name" in target_details:
+            used_mcu_names.add(target_details["device_name"])
+    return used_mcu_names
+
+
+@cmsis_mcu_descr.command(
+    short_help="Reload the cache of CMSIS MCU descriptions.  This can take several minutes."
+)
+def reload_cache():
+    """
+    Reload the cache of CMSIS MCU descriptions.  This can take several minutes.
+    Note that it's possible for various MCU vendors' CMSIS pack servers to be down, and
+    cmsis-pack-manager does not report any errors in this case (augh whyyyyy).
+
+    So, if the target you are looking for does not exist after running this command, you might
+    just have to try again the next day.  It's happened to me several times...
+    """
+    cmsis_cache = open_cmsis_cache(False)
+
+    LOGGER.info("Cleaning and redownloading CMSIS device descriptions, this may take some time...")
+    cmsis_cache.cache_clean()
+    cmsis_cache.cache_descriptors()
+
+
+@cmsis_mcu_descr.command(
+    name="find-unused",
+    short_help="Find MCU descriptions that are not used by targets.json5."
+)
+def find_unused():
+    """
+    Remove MCU descriptions that are not used by targets.json5.
+    Use this command after removing targets from Mbed to clean up old MCU definitions.
+    """
+    used_mcu_names = get_mcu_names_used_by_targets_json5()
+
+    # Accumulate set of all keys in cmsis_mcu_descriptions.json
+    LOGGER.info("Scanning cmsis_mcu_descriptions.json for MCUs to be pruned...")
+    cmsis_mcu_descriptions_json_contents: Dict[str, Any] = decode_json_file(CMSIS_MCU_DESCRIPTIONS_JSON_PATH)
+    available_mcu_names = set(cmsis_mcu_descriptions_json_contents.keys())
```

Comment:
`.keys()` is already set-like, so no need to convert it again. (also in [`def fetch_missing():` on line 142](https://github.com/mbed-ce/mbed-os/pull/282/files#diff-7454050ce86e87541f8ef1ac97c179461396323c48d86dd744cdb11aa12cb883R142))

```suggestion
    available_mcu_names = cmsis_mcu_descriptions_json_contents.keys()
```